packages/starknet-snap/src/index.tsx

```diff
@@ -302,42 +297,38 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
 };
 
 export const onInstall: OnInstallHandler = async () => {
-  await snap.request({
-    method: 'snap_dialog',
-    params: {
-      type: 'alert',
-      content: (
-        <Box>
-          <Text>Your MetaMask wallet is now compatible with Starknet!</Text>
-          <Text>
-            To manage your Starknet account and send and receive funds, visit
-            the <Link href={getDappUrl()}>companion dapp for Starknet</Link>.
-          </Text>
-        </Box>
-      ),
-    },
-  });
+  const initSnapStateManager = new InitSnapStateManager();
+  await initSnapStateManager.ensureJsxSupport(
```

Comment:
IMO, we should use a helper, rather than a state manager, which out of the purpose for managing state

1. Check if JSX supported by the utils
2. if true, then update the state
3. if false, then display a dialog

and i think we dont need to store in state, as it is TEMP (eventually browser should use the latest MM version)

we can just check it in onInstall, onUpdate, onHomepage

for standard request, just let it fail